tests/components/tasmota/test_device_trigger.py

```diff
@@ -0,0 +1,832 @@
+            return True
+
+    with patch(
+        "hatasmota.discovery.get_switch_triggers",
+        return_value=[
+            FakeTrigger(
+                event=None,
+                idx=1,
+                mac=None,
+                source=None,
+                subtype=None,
+                trigger_topic=None,
+                type=None,
+            )
+        ],
+    ):
+        async_fire_mqtt_message(
+            hass, f"{DEFAULT_PREFIX}/{mac}/config", json.dumps(config)
+        )
+        await hass.async_block_till_done()
+
+    device_entry = device_reg.async_get_device(set(), {("mac", mac)})
+    triggers = await async_get_device_automations(hass, "trigger", device_entry.id)
+    assert_lists_same(triggers, [])
+
+    # Rediscover without exception
+    async_fire_mqtt_message(hass, f"{DEFAULT_PREFIX}/{mac}/config", json.dumps(config))
+    await hass.async_block_till_done()
+
+    expected_triggers = [
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "device_id": device_entry.id,
+            "discovery_id": "00000049A3BC_switch_1_TOGGLE",
+            "type": "button_short_press",
+            "subtype": "switch_1",
+        },
+    ]
+    triggers = await async_get_device_automations(hass, "trigger", device_entry.id)
+    assert_lists_same(triggers, expected_triggers)
+
+
+async def test_update_remove_triggers(
+    hass, device_reg, entity_reg, mqtt_mock, setup_tasmota
+):
+    """Test triggers can be updated and removed."""
+    # Discover a device with toggle + hold trigger
+    config1 = copy.deepcopy(DEFAULT_CONFIG)
+    config1["swc"][0] = 5
+    mac = config1["mac"]
+
+    # Discover a device with toggle + double press trigger
+    config2 = copy.deepcopy(DEFAULT_CONFIG)
+    config2["swc"][0] = 8
+
+    # Discover a device with no trigger
+    config3 = copy.deepcopy(DEFAULT_CONFIG)
+    config3["swc"][0] = -1
+
+    async_fire_mqtt_message(hass, f"{DEFAULT_PREFIX}/{mac}/config", json.dumps(config1))
+    await hass.async_block_till_done()
+
+    device_entry = device_reg.async_get_device(set(), {("mac", mac)})
+
+    expected_triggers1 = [
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "device_id": device_entry.id,
+            "discovery_id": "00000049A3BC_switch_1_TOGGLE",
+            "type": "button_short_press",
+            "subtype": "switch_1",
+        },
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "device_id": device_entry.id,
+            "discovery_id": "00000049A3BC_switch_1_HOLD",
+            "type": "button_long_press",
+            "subtype": "switch_1",
+        },
+    ]
+    expected_triggers2 = copy.deepcopy(expected_triggers1)
+    expected_triggers2[1]["type"] = "button_double_press"
+
+    triggers = await async_get_device_automations(hass, "trigger", device_entry.id)
+    for expected in expected_triggers1:
+        assert expected in triggers
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)